src/compat/posix/idx/pipe_pty.c

```diff
@@ -275,6 +276,8 @@ static int pty_master_status(struct idesc *idesc, int mask) {
 		return 1;
 	}
 
+	mutex_lock(&t->lock);
+
 	switch (mask) {
 	case POLLIN:
 		res = ring_can_read(&pty_to_tty(pty)->o_ring, TTY_IO_BUFF_SZ, 1);
```

Comment:
&t->o_ring looks better than &pty_to_tty(pty)->o_ring

Reply:
Thank you, fixed.